src/validation.h

```diff
@@ -181,12 +181,41 @@ void UnlinkPrunedFiles(const std::set<int>& setFilesToPrune);
 /** Prune block files up to a given height */
 void PruneBlockFilesManual(int nManualPruneHeight);
 
-/** (try to) add transaction to memory pool
- * plTxnReplaced will be appended to with all transactions replaced from mempool
- * @param[out] fee_out optional argument to return tx fee to the caller **/
-bool AcceptToMemoryPool(CTxMemPool& pool, TxValidationState &state, const CTransactionRef &tx,
-                        std::list<CTransactionRef>* plTxnReplaced,
-                        bool bypass_limits, bool test_accept=false, CAmount* fee_out=nullptr) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+/**
+* Validation result for a single transaction mempool acceptance.
+* When m_accepted = true, m_replaced_transactions contains a list
+* of replaced transactions and m_base_fees contains the tx fees.
+* When m_accepted = nullopt, the transaction was not fully validated
+* (may be valid or invalid), e.g. when it is part of a package that
+* failed validation for an earlier transaction.
+*/
+struct MempoolAcceptResult {
+    // Tri-state
+    std::optional<bool> m_accepted;
+    TxValidationState m_state;
+
+    // Valid when m_accepted = true
+    std::optional<std::list<CTransactionRef>> m_replaced_transactions;
+    std::optional<CAmount> m_base_fees;
```

Comment:
abc6ff1

As this interface aims to be consumed by higher users, maybe comment those variables, like "evicted transactions per BIP 125 rules", "Raw base fees expressed in satoshis".

Reply:
Added comments

Reply:
Can you added the unit in comment (satoshi). I swear once you start to deal with fee computation you spend your time thinking if you're dealing with apples or oranges :)